src/main/java/org/elasticsearch/script/CompiledScript.java

```diff
@@ -1,43 +0,0 @@
-/*
- * Licensed to Elasticsearch under one or more contributor
- * license agreements. See the NOTICE file distributed with
- * this work for additional information regarding copyright
- * ownership. Elasticsearch licenses this file to you under
- * the Apache License, Version 2.0 (the "License"); you may
- * not use this file except in compliance with the License.
- * You may obtain a copy of the License at
- *
- *    http://www.apache.org/licenses/LICENSE-2.0
- *
- * Unless required by applicable law or agreed to in writing,
- * software distributed under the License is distributed on an
- * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
- * KIND, either express or implied.  See the License for the
- * specific language governing permissions and limitations
- * under the License.
- */
-
-package org.elasticsearch.script;
-
-/**
- *
- */
-public class CompiledScript {
```

Comment:
I thought about this for a while, and my reasoning for using a single Script in this fashion is the way caching works makes using both awkward.  Wrapping the Script object in a CompiledScript object doesn't work well because there are parameters that just don't belong there when the CompiledScript objects are cached such as ScriptContext and params because these can potentially change between compile calls.  I've tried to compromise a bit here because I do also like the separation between the two.  I've added back the CompiledScript object and renamed the Script object to ScriptParams.  In the CompiledScript object I've added the parameters name and script type since these should not change ever.  For this to work, however, I had to modify the validation method to ensure that an inline script would not be cached in place of an indexed script or the name for the indexed script would be lost.